src/Microsoft.DotNet.Build.Tasks/GenerateCurrentVersion.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Xml;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class GenerateCurrentVersion : Task
+    {
+        /// <summary>
+        /// The path to the file that contains the date of the last comit in EPOCH time.
+        /// </summary>
+        [Required]
+        public string CurrentDate { get; set; }
+
+        /// <summary>
+        /// The Major Version that will be used for this build.
+        /// </summary>
+        [Output]
+        public string ResultCurrentVersion { get; set; }
+
+        public override bool Execute()
+        {
+            DateTime date;
+            ResultCurrentVersion = string.Empty;
+            bool isValidDate = DateTime.TryParse(CurrentDate, out date);
+            if (!isValidDate)
+            {
+                Log.LogError("The date '{0}' is not valid. Please specify a date in a valid DateTime format.", CurrentDate);
+                return false;
+            }
+            ResultCurrentVersion = GetCurrentVersionForDate(date);
+            if (string.IsNullOrEmpty(ResultCurrentVersion))
+            {
+                Log.LogError("The date '{0}' is not valid. Please pass in a date after April 1996.", CurrentDate);
+                return false;
+            }
+            return true;
+        }
+
+        public static string GetCurrentVersionForDate(DateTime date)
+        {
+            DateTime compareDate = new DateTime(1996,4,1);
```

Comment:
You should add a comment in the code. Something along the lines of "We need to ensure our build numbers are higher then what we used to ship from our internal builds so we are using a date to align with that number". I also think this should be an optional parameter which we can default to this as I imagine there will be folks using this that don't have that constraint and will want to start the with a lower date.